File: lang/python/src/fun/vac/algorithms/number_theory/Coprimality.py
from fun.vac.algorithms.number_theory import GreatestCommonDivisor

def reduceToBinaryGCD(m: int, n: int) -> int:
    return GreatestCommonDivisor.iterativeBinaryGCD(m, n) == 1

def reduceToEuclidean(m: int, n: int) -> int:
    return GreatestCommonDivisor.iterativeEuclidean(m, n) == 1

File: lang/ruby/src/fun/vac/algorithms/number_theory/Coprimality.rb
require "fun/vac/algorithms/number_theory/GreatestCommonDivisor"

module Coprimality

    def self.reduceToBinaryGCD(m, n)
        GreatestCommonDivisor.iterativeBinaryGCD(m, n) == 1
    end

    def self.reduceToEuclidean(m, n)
        GreatestCommonDivisor.iterativeEuclidean(m, n) == 1
    end
end
